external-config/src/main/groovy/grails/plugin/externalconfig/ExternalConfigRunListener.groovy

```diff
@@ -138,15 +138,15 @@ class ExternalConfigRunListener implements SpringApplicationRunListener {
         WriteFilteringMap filterMap = new WriteFilteringMap(currentConfig)
         slurper.binding = filterMap
         ConfigObject configObject = slurper.parse(configText)
-        Map properties = configText ? configObject?.flatten() : [:]
+        Map<String, Object> properties = configText ? configObject?.flatten() as Map<String, Object> : [:]
         Map writtenValues = filterMap.getWrittenValues()
         properties.putAll(writtenValues)
         new MapPropertySource(resource.filename, properties)
     }
 
     private NavigableMapPropertySource loadYamlConfig(Resource resource) {
         log.info("Loading YAML config file {}", resource.URI)
-        NavigableMapPropertySource propertySource = yamlPropertySourceLoader.load(resource.filename, resource, null) as NavigableMapPropertySource
+        NavigableMapPropertySource propertySource = yamlPropertySourceLoader.load(resource.filename, resource, null)?.first() as NavigableMapPropertySource
```

Comment:
Need a second opinion if this is ok - seems to have changed to a list of `PropertySource`'s - but we're only passing in one resource so grabbing the first may be ok?

Reply:
And it is not because `yml` resources are considered a "next" resource when it encounters the separator `----` ?  I think a test case should prove this.

Reply:
Makes sense, will add test.